File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/sed/medlemskap/impl/MedlemskapA010.java
package no.nav.melosys.eessi.models.sed.medlemskap.impl;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.EqualsAndHashCode;
import no.nav.melosys.eessi.models.sed.medlemskap.Medlemskap;
import no.nav.melosys.eessi.models.sed.nav.MeldingOmLovvalg;
import no.nav.melosys.eessi.models.sed.nav.Utsendingsland;
import no.nav.melosys.eessi.models.sed.nav.VedtakA010;

@EqualsAndHashCode(callSuper = false)
@JsonIgnoreProperties(ignoreUnknown = true)
@Data
public class MedlemskapA010 implements Medlemskap {

    private Utsendingsland andreland;

    private VedtakA010 vedtak;

    private MeldingOmLovvalg meldingomlovvalg;
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/sed/medlemskap/impl/MedlemskapA010.kt
package no.nav.melosys.eessi.models.sed.medlemskap.impl

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import no.nav.melosys.eessi.models.sed.medlemskap.Medlemskap
import no.nav.melosys.eessi.models.sed.nav.MeldingOmLovvalg
import no.nav.melosys.eessi.models.sed.nav.Utsendingsland
import no.nav.melosys.eessi.models.sed.nav.VedtakA010

@JsonIgnoreProperties(ignoreUnknown = true)
data class MedlemskapA010(
    var andreland: Utsendingsland? = null,
    var vedtak: VedtakA010? = null,
    var meldingomlovvalg: MeldingOmLovvalg? = null
) : Medlemskap
